workers/unity/Packages/io.improbable.gdk.debug/NetStatViewer.cs

```diff
@@ -0,0 +1,164 @@
+        private const string UpdateRowUxmlPath = "Packages/io.improbable.gdk.debug/Templates/UpdateRow.uxml";
+
+        private World selectedWorld;
+        private NetworkStatisticsSystem netStatSystem;
+        private List<(string Name, uint ComponentId)> spatialComponents;
+        private Dictionary<int, VisualElement> listElements;
+
+        [MenuItem("SpatialOS/Window/Network Analyzer", false)]
+        public static void ShowWindow()
+        {
+            var inspectorWindowType = typeof(EditorWindow).Assembly.GetType("UnityEditor.InspectorWindow");
+            var window = GetWindow<NetStatViewer>(inspectorWindowType);
+            window.titleContent.text = "Network Analyzer";
+            window.Show();
+        }
+
+        private void OnEnable()
+        {
+            // Generate list of types
+            spatialComponents = ComponentDatabase.ComponentsToIds
+                .Select(pair => (pair.Key.DeclaringType.Name, pair.Value))
+                .ToList();
+
+            // Load UI
+            SetupUI();
+            SetupWorldSelection();
+        }
+
+        // Update UI at 10 FPS
+        private void OnInspectorUpdate()
+        {
+            SetupWorldSelection();
+            RefreshUpdates();
+        }
+
+        private void SetupUI()
+        {
+            // Load main window
+            var windowTemplate = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(WindowUxmlPath);
+            windowTemplate.CloneTree(rootVisualElement);
+
+            // Load update row
+            var itemTemplate = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(UpdateRowUxmlPath);
+
+            // Clone row for each component, fill list
+            var scrollView = rootVisualElement.Q<ScrollView>("updatesContainer");
+            for (var i = 0; i < spatialComponents.Count; i++)
+            {
+                var element = itemTemplate.CloneTree();
+                UpdateElement(i, element);
+                scrollView.Add(element);
+            }
+        }
+
+        private void SetupWorldSelection()
+        {
+            // Find spatial worlds
+            var spatialWorlds = World.AllWorlds.Where(w => w.GetExistingSystem<NetworkStatisticsSystem>() != null)
+                .ToList();
+
+            // Fill menu items
+            var worldMenu = rootVisualElement.Q<ToolbarMenu>("worldSelector").menu;
+            for (var i = 0; i < spatialWorlds.Count; i++)
+            {
+                var spatialWorld = spatialWorlds[i];
+                worldMenu.InsertAction(i, spatialWorld.Name,
+                    action => SelectWorld((World) action.userData),
+                    action => selectedWorld == (World) action.userData
+                        ? DropdownMenuAction.Status.Checked
+                        : DropdownMenuAction.Status.Normal,
+                    spatialWorld);
+            }
+
+            // Trim excess items
```

Comment:
Why do we need todo this?

Reply:
So that the Worlds get removed from the list after we exit play mode, or disconnect a single world.

Reply:
Ah right